build_resolvers/lib/src/analysis_driver_model.dart

```diff
@@ -167,3 +193,108 @@ extension _AssetIdExtensions on AssetId {
   /// Asset path for the in-memory filesystem.
   String get asPath => AnalysisDriverModelUriResolver.assetPath(this);
 }
+
+/// The directive graph of all known sources.
+///
+/// Also tracks whether there is a `.transitive_digest` file next to each source
+/// asset, and tracks missing files.
+class _Graph {
+  final Map<AssetId, _Node> nodes = {};
+
+  /// Walks the import graph from [ids] loading into [nodes].
+  Future<void> load(AssetReader reader, Iterable<AssetId> ids) async {
+    final nextIds = Queue.of(ids);
```

Comment:
I would have commented about this in one of the previous reviews as well but forgot.
Is there any reason this is a Queue instead of a List? A List should do `removeLast` instead - so ordering would change -but otherwise I'd assume it was similar and probably perform better because it doesn't have to do casts and head and tail book keeping etc.

Reply:
No particular reason; but any cost here should be `O(n)`, I want to solve the `O(n^2)` issues before getting onto constant factor improvements ... added a TODO. Thanks.